local_planner/src/nodes/local_planner_node.cpp

```diff
@@ -908,10 +946,17 @@ void LocalPlannerNode::threadFunction() {
                 (std::clock() - start_time) / (double)(CLOCKS_PER_SEC / 1000));
     }
 
-    // publish log name
-    std_msgs::String msg;
-    msg.data = local_planner_.log_name_;
-    log_name_pub_.publish(msg);
+    // publish performance data
+    std_msgs::String cpu_msg;
+    std_msgs::String mem_msg;
+    std::string cpu = exec(" top -bn 1 | awk '{print $9}' | tail -n +8 | awk '{s+=$1} END {print s/8}'");
```

Comment:
CPU & Memory monitoring should IMO not be part of the planner node.
There are already existing system monitoring nodes for ROS, for example https://github.com/ethz-asl/ros-system-monitor. Is there any reason not to use that?

Reply:
sounds good, I will look into that